civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -46,17 +44,14 @@
          * @param {string} mode
          */
         function modalDocument (id, role, mode) {
-          $rootScope.$broadcast('ct-spinner-show', 'document-' + id);
-          vm.loadingModalData = true;
-
           DocumentService.get({ id: id })
             .then(function(data) {
               if (!data) {
                 throw new Error('Requested Document is not available');
               }
-
+              vm.openModalDocument(data[0], role, mode);
               vm.cacheContacts(data).then(function () {
-                vm.openModalDocument(data[0], role, mode);
+                $rootScope.isLoading = false;
```

Comment:
why are you using `$rootScope`?

Reply:
Since `isLoading` is used to display or hide the loader in document modal. In case, when the document modal is loaded form SSP and if the `isLoading` is bound to `vm` in SSP 
it cannot be accessed in modal controller  T&A and vice-versa. So I have used `$rootScope` so, that the changes to `isLoading` in either SSP or T&A  will be reflected in document modal.